### ISSUE TITLE
Persistent Julia engine for chunk evaluation

### ISSUE THREAD

**Issue by yutannihilation** [rstudio/rstudio#1798]
(Julia version of #1651)

knitr will soon support Julia chunks with JuliaCall package. I don't have strong opinion on the support for Julia, but it would be useful if we can use Julia in RStudio seamlessly as well as Python.

**Comment by pbhogale** [comment]
strongly second this. Julia is going to be incredibly important to the numerical computing community in years to come, and already has a vibrant, friendly community. would be great to be able to work in Rmd with seamless exchange between R and Julia, and be able to seamlessly import julia functions into R scripts a la reticulate and python.

**Comment by harryprince** [comment]
Julia is pretty powerful in AIOT era, programming like Python and running like C.
Wish RStudio can support Julia code intelligence.
Julia is good at optimization problem. JuMP eco-system is pretty mature like Stan in R and Tensorflow in Python.

**Comment by Tokazama** [comment]
Doesn't RStudio use Ace for syntax highlighting? I'm pretty sure they already have syntax highlighting for Julia. Even that would be nice.

**Comment by longhua8800w** [comment]
strongly second this.

**Comment by Tokazama** [comment]
This doesn't appear to be a high priority  RStudio team at the moment so people might want to look to https://github.com/julia-vscode/julia-vscode instead.

**Comment by pbhogale** [comment]
How hard would it be to code an add-in to RStudio that adds Julia syntax highlighting ? (I genuinely don't know, if someone points me in the right direction I'm happy to give it a try)

**Comment by jmcphers** [comment]
It already has it! :-)

![image](https://user-images.githubusercontent.com/470418/96480825-99cae900-11ef-11eb-8c4d-c3a4abbe663e.png)

**Comment by rleyvasal** [comment]
I also want to see support for Julia in RStudio.  
Two things I want to be able to do  are:

![julia_chunk](https://user-images.githubusercontent.com/61335136/111078100-32b91400-84b1-11eb-8c24-9c33ef86415d.PNG)
1. I want to see the Julia variables created in code chunks in the Global Environment on the top right of RStudio (same as R and Python show). 

2. When using  code chunks in a R notebook, I want to be able to run one line of code at a time inside the Julia chunk (right now ctrl+enter runs the whole chunk)

**Comment by kevinushey** [comment]
We could probably use https://github.com/stefan-m-lenz/JuliaConnectoR to support this. We could even rely on a knitr engine implementation using that package.

**Comment by sourish-cmi** [comment]
This will be great to have a feature in RStudio.

**Comment by PharmCat** [comment]
it will be very useful for R and Julia community

**Comment by rcberg** [comment]
Please!!!

**Comment by shawndoyle** [comment]
Agreed, this would be excellent.

**Comment by ParadaCarleton** [comment]
> We could probably use https://github.com/stefan-m-lenz/JuliaConnectoR to support this. We could even rely on a knitr engine implementation using that package.

Just a quick note--in cases where it works, [JuliaCall](https://github.com/Non-Contradiction/JuliaCall) is typically much faster.

**Comment by aamangat** [comment]
Yes I strongly recommend to have Julia in Rstudio.

**Comment by Zhaoju-Deng** [comment]
for data analysis, please integrate julia in rstudio, that's much more sophisticated than in vscode

**Comment by studsttat** [comment]
Yes, I strongly second to this. It will be astonishing to have Julia in RStudio and you can even source a Julia (.jl) script files just like Python.
